### ISSUE TITLE
Training performance

### ISSUE THREAD

**Issue by Thocle** [mk-runner/MLRG#14]
Hello, first thanks for your work on RRG.
When I reproduce the paper, the metrics seem to be a little bit lower than expected, like test_bleu_4 is 0.13419528549355245. Could I ask where the potential problems might be?

env(different from yours)
python 3.10.18
CUDA 12.1
torch 2.3.1

And I found there are already some differences in stage 1 like higher val loss. Especially the initial val loss is higher. Is this the origin of the problems?

```
task:	pretrain
data_name:	mimic_cxr
ann_path:	/data4/MLRG/data/five_work_mimic_cxr_annotation_v1.1.json
view_position_embed:	/data4/MLRG/data/five_work_mimic_cxr_view_position_v1.1.json
images_dir:	/data1/data/mimic_cxr_jpg/physionet.org/files/mimic-cxr-jpg/2.0.0/files/
max_length:	100
num_workers:	12
is_save_checkpoint:	True
is_multiview_learning:	True
is_prior_scan:	True
using_mpc_loss:	True
using_local_loss:	False
is_prior_report:	True
is_indication:	True
report_style:	factual_serialization
ckpt_zoo_dir:	/data1/checkpoints/
text_encoder_num_layers:	6
cross_modal_fusion_num_layers:	1
multiview_fusion_num_layers:	3
num_heads:	8
pt_lr:	5e-05
ft_lr:	5e-05
temp:	0.5
monitor_metric:	val_epoch_loss
epochs:	20
batch_size:	32
hidden_size:	768
resume:	None
load:	None
test_ckpt_path:	/home/miao/data/Code/five_gpt2/script/results/mimic_cxr/finetune/v1011-MLRG-ft-RCB_2024_10_12_14/best_model.ckpt
version:	v0906_fs_pt
chexbert_path:	/data1/checkpoints/chexbert.pth
bert_path:	/data1/checkpoints/bert-base-uncased
rad_dino_path:	/data1/checkpoints/rad-dino
radgraph_path:	/data1/checkpoints/radgraph
cxr_bert_path:	/data1/checkpoints/BiomedVLP-CXR-BERT-specialized
distilgpt2_path:	/data1/checkpoints/distilgpt2
cvt2distilgpt2_path:	/data1/checkpoints/mimic_cxr_jpg_chen/cvt_21_to_gpt2/epoch=8-val_chen_cider=0.425092.ckpt
seed:	9233
num_gpus:	1
num_beams:	3
save_period:	1
exp_dir_trial:	results/mimic_cxr/pretrain/v0906_fs_pt_2025_10_10_19
print_step:	500
device:	cuda
monitor_mode:	min
checkpoint_dir:	results/mimic_cxr/pretrain/v0906_fs_pt_2025_10_10_19/checkpoint
time:	2025_10_10_19

No. of training & validation examples: 240197 & 2113.
Epoch 0, validation step 0/67, instance_loss = 3.47, mpc_loss = 3.63, loss = 7.09, lr: 5e-05
###############################################################
Epoch 0, Validation is over, current val loss:loss = 6.83, mpc_loss = 3.37, instance_loss = 3.47, lr: 5e-05
best validation loss: loss = 6.833403587341309, mpc_loss = 3.3672168254852295, instance_loss = 3.4661865234375, epoch = 0

Epoch 0, training step 0/7506, instance_loss = 3.46, mpc_loss = 4.2, loss = 7.66, lr: 5e-05
Epoch 0, training step 500/7506, instance_loss = 2.99, mpc_loss = 3.03, loss = 6.01, lr: 5e-05
Epoch 0, training step 1000/7506, instance_loss = 2.82, mpc_loss = 2.62, loss = 5.43, lr: 5e-05
Epoch 0, training step 1500/7506, instance_loss = 2.83, mpc_loss = 2.7, loss = 5.53, lr: 5e-05
Epoch 0, training step 2000/7506, instance_loss = 2.63, mpc_loss = 2.46, loss = 5.08, lr: 5e-05
Epoch 0, training step 2500/7506, instance_loss = 2.61, mpc_loss = 2.61, loss = 5.22, lr: 5e-05
Epoch 0, training step 3000/7506, instance_loss = 2.5, mpc_loss = 2.68, loss = 5.18, lr: 5e-05
Epoch 0, training step 3500/7506, instance_loss = 2.55, mpc_loss = 2.22, loss = 4.77, lr: 5e-05
Epoch 0, training step 4000/7506, instance_loss = 2.28, mpc_loss = 2.43, loss = 4.71, lr: 5e-05
Epoch 0, training step 4500/7506, instance_loss = 2.51, mpc_loss = 2.57, loss = 5.08, lr: 5e-05
Epoch 0, training step 5000/7506, instance_loss = 2.57, mpc_loss = 2.5, loss = 5.07, lr: 5e-05
Epoch 0, training step 5500/7506, instance_loss = 2.48, mpc_loss = 2.32, loss = 4.8, lr: 5e-05
Epoch 0, training step 6000/7506, instance_loss = 2.56, mpc_loss = 2.58, loss = 5.14, lr: 5e-05
Epoch 0, training step 6500/7506, instance_loss = 2.44, mpc_loss = 2.4, loss = 4.85, lr: 5e-05
Epoch 0, training step 7000/7506, instance_loss = 2.3, mpc_loss = 2.44, loss = 4.74, lr: 5e-05
Epoch 0, training step 7500/7506, instance_loss = 2.25, mpc_loss = 2.37, loss = 4.62, lr: 5e-05
Epoch 0, training step 7505/7506, instance_loss = 2.54, mpc_loss = 2.41, loss = 4.95, lr: 5e-05
Epoch 0, validation step 0/67, instance_loss = 2.37, mpc_loss = 2.0, loss = 4.36, lr: 5e-05
Epoch 0, validation step 66/67, instance_loss = -0.0, mpc_loss = -0.0, loss = -0.0, lr: 5e-05
###############################################################
Epoch 0, Validation is over, current val loss:loss = 4.77, mpc_loss = 2.12, instance_loss = 2.65, lr: 5e-05
best validation loss: loss = 4.769687652587891, mpc_loss = 2.1163454055786133, instance_loss = 2.6533422470092773, epoch = 0

Epoch 0, Training is over, loss = 5.18, mpc_loss = 2.58, instance_loss = 2.6, lr: 5e-05
###############################################################
Epoch 1, training step 0/7506, instance_loss = 2.39, mpc_loss = 2.27, loss = 4.65, lr: 5e-05
Epoch 1, training step 500/7506, instance_loss = 2.44, mpc_loss = 2.51, loss = 4.94, lr: 5e-05
Epoch 1, training step 1000/7506, instance_loss = 2.23, mpc_loss = 2.46, loss = 4.7, lr: 5e-05
Epoch 1, training step 1500/7506, instance_loss = 2.4, mpc_loss = 2.22, loss = 4.62, lr: 5e-05
Epoch 1, training step 2000/7506, instance_loss = 2.25, mpc_loss = 2.46, loss = 4.71, lr: 5e-05
Epoch 1, training step 2500/7506, instance_loss = 2.44, mpc_loss = 2.51, loss = 4.96, lr: 5e-05
Epoch 1, training step 3000/7506, instance_loss = 2.38, mpc_loss = 2.46, loss = 4.84, lr: 5e-05
Epoch 1, training step 3500/7506, instance_loss = 2.44, mpc_loss = 2.45, loss = 4.89, lr: 5e-05
Epoch 1, training step 4000/7506, instance_loss = 2.4, mpc_loss = 2.27, loss = 4.67, lr: 5e-05
Epoch 1, training step 4500/7506, instance_loss = 2.64, mpc_loss = 2.45, loss = 5.09, lr: 5e-05
Epoch 1, training step 5000/7506, instance_loss = 2.35, mpc_loss = 2.53, loss = 4.88, lr: 5e-05
Epoch 1, training step 5500/7506, instance_loss = 2.37, mpc_loss = 2.31, loss = 4.68, lr: 5e-05
Epoch 1, training step 6000/7506, instance_loss = 2.29, mpc_loss = 2.38, loss = 4.67, lr: 5e-05
Epoch 1, training step 6500/7506, instance_loss = 2.21, mpc_loss = 2.33, loss = 4.53, lr: 5e-05
Epoch 1, training step 7000/7506, instance_loss = 2.28, mpc_loss = 2.47, loss = 4.75, lr: 5e-05
Epoch 1, training step 7500/7506, instance_loss = 2.23, mpc_loss = 2.43, loss = 4.66, lr: 5e-05
Epoch 1, training step 7505/7506, instance_loss = 2.39, mpc_loss = 2.37, loss = 4.75, lr: 5e-05
Epoch 1, validation step 0/67, instance_loss = 2.37, mpc_loss = 1.95, loss = 4.32, lr: 5e-05
Epoch 1, validation step 66/67, instance_loss = -0.0, mpc_loss = -0.0, loss = -0.0, lr: 5e-05
###############################################################
Epoch 1, Validation is over, current val loss:loss = 4.7, mpc_loss = 2.09, instance_loss = 2.61, lr: 5e-05
best validation loss: loss = 4.701042175292969, mpc_loss = 2.0879790782928467, instance_loss = 2.613062620162964, epoch = 1

Epoch 1, Training is over, loss = 4.76, mpc_loss = 2.38, instance_loss = 2.38, lr: 5e-05
###############################################################
Epoch 2, training step 0/7506, instance_loss = 2.33, mpc_loss = 2.46, loss = 4.78, lr: 5e-05
Epoch 2, training step 500/7506, instance_loss = 2.35, mpc_loss = 2.19, loss = 4.53, lr: 5e-05
......

```
